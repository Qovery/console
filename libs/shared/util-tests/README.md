# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
