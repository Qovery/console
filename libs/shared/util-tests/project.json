{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
