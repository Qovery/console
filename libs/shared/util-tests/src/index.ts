export * from './lib/shared-util-tests'
