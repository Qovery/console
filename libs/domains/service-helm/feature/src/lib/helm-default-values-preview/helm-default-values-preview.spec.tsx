import { renderWithProviders } from '@qovery/shared/util-tests'
import HelmDefaultValuesPreview from './helm-default-values-preview'

jest.mock('../hooks/use-helm-default-values/use-helm-default-values', () => {
  return {
    ...jest.requireActual('../hooks/use-helm-default-values/use-helm-default-values'),
    useHelmDefaultValues: () => ({
      data: 'my-yaml-content',
      isLoading: true,
    }),
  }
})

describe('HelmDefaultValuesPreview', () => {
  it('should match snapshot', async () => {
    const { baseElement } = renderWithProviders(<HelmDefaultValuesPreview />)
    expect(baseElement).toBeTruthy()
  })
})
