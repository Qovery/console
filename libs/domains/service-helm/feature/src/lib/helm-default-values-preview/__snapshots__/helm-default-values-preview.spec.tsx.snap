// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelmDefaultValuesPreview should match snapshot 1`] = `
<body>
  <div>
    <section
      style="display: flex; position: relative; text-align: initial; width: 100%; height: 100vh;"
    >
      <div
        style="display: flex; height: 100%; width: 100%; justify-content: center; align-items: center;"
      >
        <div
          class="aspect-square border-2  rounded-full border-solid border-r-neutral-500 animate-spin w-4 "
          data-testid="spinner"
        />
      </div>
      <div
        style="width: 100%; display: none;"
      />
    </section>
  </div>
  <script
    src="https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0/min/vs/loader.js"
  />
</body>
`;
